src/main/src/main/java/org/geoserver/security/impl/DefaultResourceAccessManager.java

```diff
@@ -493,12 +506,26 @@ private String[] getLayerGroupPath(LayerGroupInfo layerGroup) {
 
     @Override
     public Filter getSecurityFilter(Authentication user, Class<? extends CatalogInfo> clazz) {
+        Filter filter = filterCache.getIfPresent(Pair.of(user, clazz));
+        if (filter == null) {
+            filter = getSecurityFilterInternal(user, clazz);
+            if (filter != null) {
+                filterCache.put(Pair.of(user, clazz), filter);
+            }
+        }
```

Comment:
Please use `V get(K key, Callable<? extends V> loader)` instead, it'll handle concurrency for you.
e.g.
```java
Filter filter = filterCache.get(Pair.of(user, clazz)), ()-> getSecurityFilterInternal(user, clazz))
```

Reply:
This will fail if getSecurityFilterInternal returns null. I didn't want to store post filters in the cache, that is why I create them only if getSecurityFilterInternal returns null